kythe/go/platform/tools/kzip/mergecmd/mergecmd.go

```diff
@@ -50,24 +52,49 @@ func New() subcommands.Command {
 // for merging kzip files.
 func (c *mergeCommand) SetFlags(fs *flag.FlagSet) {
 	fs.StringVar(&c.output, "output", "", "Path to output kzip file")
+	fs.BoolVar(&c.append, "append", false, "Whether to additionally merge the contents of the existing output file, if it exists")
 }
 
 // Execute implements the subcommands interface and merges the provided files.
 func (c *mergeCommand) Execute(ctx context.Context, fs *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
 	if c.output == "" {
 		return c.Fail("required --output path missing")
 	}
-	if err := mergeArchives(ctx, c.output, fs.Args()); err != nil {
-		return c.Fail("Error merging archives: %v", err)
+	dir, file := filepath.Split(c.output)
+	if dir == "" {
+		dir = "."
+	}
+	tmpOut, err := vfs.CreateTempFile(ctx, dir, file)
+	tmpName := tmpOut.Name()
+	defer func() {
+		if tmpOut != nil {
+			tmpOut.Close()
+			vfs.Remove(ctx, tmpName)
+		}
+	}()
+	if err != nil {
+		return c.Fail("Error creating temp output: ", err)
+	}
+	archives := fs.Args()
+	if c.append {
+		orig, err := vfs.Open(ctx, c.output)
+		if err == nil {
+			archives = append([]string{c.output}, archives...)
+			if err := orig.Close(); err != nil {
+				return c.Fail("Error closing original: ", err)
+			}
+		}
+	}
+	if err := mergeArchives(ctx, tmpOut, archives); err != nil {
+		return c.Fail("Error merging archives: ", err)
+	}
+	if err := vfs.Rename(ctx, tmpName, c.output); err != nil {
+		return c.Fail("Error renaming tmp to output:", err)
 	}
 	return subcommands.ExitSuccess
 }
 
-func mergeArchives(ctx context.Context, output string, archives []string) error {
-	out, err := vfs.Create(ctx, output)
-	if err != nil {
-		return fmt.Errorf("error creating output: %v", err)
-	}
+func mergeArchives(ctx context.Context, out vfs.TempFile, archives []string) error {
```

Comment:
`out` should just be typed `io.WriteCloser`.

Reply:
Or pass in the kzip Writer

Reply:
Prefer to pass in the WriteCloser.

done